src/excmds.ts

```diff
@@ -2664,7 +2666,191 @@ export async function viewcontainers() {
+    if (name == await windowTgroup()) {
+        throw new Error(`Already on tab group "${name}"`)
+    }
+
+    const groups = await tgroups()
+    if (groups.size > 0) {
+        if (groups.has(name)) {
+            return tgroupActivate(name).then(() => undefined)
+        } else {
+            return tgroupcreate(name).then(() => undefined)
+        }
+    } else {
+        return tgroupcreate(name).then(() => undefined)
+    }
+}
+
+/**
+ * Switch to the previously active tab group.
+ */
+//#background
+export async function tgrouplast() {
+    if ((await tgroups()).size < 2) {
+        throw new Error("No last tab group")
+    }
+
+    return tgroupActivateLast().then(() => undefined)
+}
+
+/**
+ * Rename the current tab group.
+ *
+ * @param name The new name of the tab group.
+ *
+ */
+//#background
+export async function tgrouprename(name: string) {
+    if ((await tgroups()).size == 0) {
+        throw new Error("No tab groups exist")
+    }
+
+    return tgroupClearOldInfo(await windowTgroup() as string, name).then(() => undefined)
+}
+
+/**
+ * Close the current tab group.
+ *
+ * First switch to the previously active tab group. Do nothing if there is only
+ * one tab group.
+ *
+ */
+//#background
+export async function tgroupclose() {
+    const groups = await tgroups()
+    if (groups.size == 0) {
+        throw new Error("No tab groups exist")
+    } else if (groups.size == 1) {
+        throw new Error("This is the only tab group")
+    } else if (groups.size > 1) {
+        const closeGroup = await windowTgroup() as string
+        await tgroupActivateLast()
+        await tgroupRun(closeGroup).then(tabs => browser.tabs.remove(tabs.map(tab => tab.id)))
+        return tgroupClearOldInfo(closeGroup).then(() => undefined)
+    }
+}
+
+/**
+ * Move the current tab to another tab group.
+ *
+ * @param name The name of the tab group to move the tab to.
+ *
+ * If this is the last tab in the tab group, also switch to tab group, keeping
+ * the current tab active.
+ *
+ */
+//#background
+export async function tgroupmove(name: string) {
+    const groups = await tgroups()
+    const currentGroup = await windowTgroup() as string
+
+    if (groups.size == 0) {
+        throw new Error("No tab groups exist")
+    }
+    if (!groups.has(name)) {
+        throw new Error(`Tab group "${name}" does not exist`)
+    }
+    if (name == currentGroup) {
+        throw new Error(`Tab is already on group "${name}"`)
+    }
+
+    const tabCount = await tgroupRun(currentGroup).then(tabs => tabs.length)
+
+    await setTabTgroup(name)
+    const currentTabId = await activeTabId()
+
+    // switch to other group if this is the last tab in the current group
+    if (tabCount == 1) {
+        return Promise.all([
+            tgroupClearOldInfo(currentGroup),
+            setWindowTgroup(name),
+            setTabTgroup(name, currentTabId),
+            tgroupRun(name).then(tabs => browserBg.tabs.show(tabs.map(tab => tab.id))),
```

Comment:
Wouldn't it be sufficient to call `tgroupClearOldInfo(currentGroup, name)` here? Or does this do something more?

Reply:
Yes, updated. Still need to show the new tabs though.